packages/safe-ds-lang/src/language/generation/safe-ds-python-generator.ts

```diff
@@ -100,7 +111,7 @@ export class SafeDsPythonGenerator {
         this.partialEvaluator = services.evaluation.PartialEvaluator;
     }
 
-    generate(document: LangiumDocument, destination: URI): TextDocument[] {
+    generate(document: LangiumDocument, destination: URI, sourceMapDestination: URI | undefined): TextDocument[] {
```

Comment:
As explained below, there is a name convention for source maps. Thus, setting the destination of the source map is not needed. However, it would be nice to have a flag to configure whether a source map should be generated or now. 

Since JS/TS does not support named arguments directly, the usual approach is to wrap such options into an object:

```ts
interface GenerateOptions {
    destination: URI,
    createSourceMaps: boolean
}
```

This clarifies the meaning of the passed arguments:

```
generate(document, { destination: ..., createSourceMaps: true })
```

Reply:
This would also allow the creation of source maps for multiple generated files in the same directory.

Reply:
Do you mean, the `GenerateOptions` interface should be moved from `cli/generate.ts` to `language/generation/safe-ds-python-generator.ts`?

Reply:
If the options are identical, yes.

Reply:
And talking about the CLI: A new flag for the `generate` command to control the creation of source maps from there would be nice, too.

Reply:
Other than the `destination` field being optional, they would be identical